CommandLine/emake/EnigmaPlugin.cpp

```diff
@@ -57,7 +57,8 @@ int EnigmaPlugin::Load()
 
   // Bind Functions
 #if CURRENT_PLATFORM_ID == OS_WINDOWS
-#	define BindFunc(x, y) GetProcAddress(static_cast<HMODULE>(x), y)
+                        // LOL fucking Winblows
+#	define BindFunc(x, y) reinterpret_cast<void*>(GetProcAddress(static_cast<HMODULE>(x), y))
```

Comment:
@JoshDreamland is this an ok/safe fix to the following warnings? I know that it does make the warnings go away...?

```
EnigmaPlugin.cpp: In member function 'int EnigmaPlugin::Load()':
EnigmaPlugin.cpp:65:96: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'const char* (*)(EnigmaCallbacks*)' [-Wcast-function-type]
   plugin_Init = reinterpret_cast<const char*(*)(EnigmaCallbacks*)>(BindFunc(_handle, "libInit"));
                                                                                                ^
```

Reply:
Better not to cast a function pointer to `void*`... what happens if you don't?

Reply:
The warning shows for each one of the binds with GCC 8.2. It's because `BindFunction` calls `GetProcAddress` on Win32 and that returns `int*` which is silly?

Reply:
@rpjohnst looked over this and suggested a c-style cast or the use of a pragma directive, but stated that the intermediate reinerpret_cast is technically fine. I don't think the c-style cast will fix it, because of other issues I am having with Clang and GCC. I don't like using a pragma to disable the warning because then I have to wrap that into an ifdef too because MSVC will warn about GCC directives. So I'm just leaving the intermediate reinterpret_cast to fix the warning.

Reply:
I did not say that. Casting a function pointer to `void*` is UB.

Reply:
Ok: dc97435018406478bdd22d3dd5f7733b0c63358f

That does fix the warnings and does not result in undefined behavior now.

I managed to actually see the error of my ways thanks to Rusky. It really doesn't matter the kind of cast you use, I just needed to be casting to a function pointer (`void(*)()`) not a regular pointer (`void*`).

This is resolved then.